trading_calendars/exchange_calendar_xbog.py

```diff
@@ -0,0 +1,201 @@
+#
+# Copyright 2019 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time
+from pandas.tseries.holiday import (
+    DateOffset,
+    Easter,
+    GoodFriday,
+    Holiday,
+    MO,
+    previous_friday,
+)
+from pandas.tseries.offsets import Day
+from pytz import timezone
+
+from .common_holidays import (
+    epiphany,
+    immaculate_conception,
+    maundy_thursday,
+    new_years_day,
+    new_years_eve,
+    european_labour_day,
+    christmas,
+)
+from .trading_calendar import (
+    TradingCalendar,
+    HolidayCalendar,
+)
+
+# Used for "first Monday on or after date"
+next_monday_offset = DateOffset(weekday=MO(1))
+
+NewYearsDay = new_years_day()
+
+Epiphany = epiphany(observance=next_monday_offset)
+
+StJosephsDay = Holiday(
+    "St. Joseph's Day (next Monday)",
+    month=3,
+    day=19,
+    offset=next_monday_offset,
+)
+
+MaundyThursday = maundy_thursday()
+
+LabourDay = european_labour_day()
+
+MondayAfterAscensionDay = Holiday(
+    "Monday After Ascension Day",
+    month=1,
+    day=1,
+    offset=[Easter(), Day(43)],
+)
+
+MondayAfterCorpusChristi = Holiday(
+    "Monday After Corpus Christi",
+    month=1,
+    day=1,
+    offset=[Easter(), Day(64)],
+)
+
+MondayAfterSacredHeart = Holiday(
+    "Monday After Sacred Heart",
+    month=1,
+    day=1,
+    offset=[Easter(), Day(71)],
+)
+
+StPeterAndStPaulDay = Holiday(
+    "St. Peter and St. Paul Day",
+    month=6,
+    day=29,
+    offset=next_monday_offset
+)
+
+ColombiaIndependenceDay = Holiday(
+    "Colombian Independence Day",
+    month=7,
+    day=20,
+)
+
+BattleOfBoyaca = Holiday(
+    "Battle of Boyaca",
+    month=8,
+    day=7,
+)
+
+AssumptionDay = Holiday(
+    "Assumption Day (next Monday)",
+    month=8,
+    day=15,
+    offset=next_monday_offset,
+)
+
+DiaDeLaRaza = Holiday(
+    'Dia de la Raza',
+    month=10,
+    day=12,
+    offset=next_monday_offset,
+)
+
+AllSaintsDay = Holiday(
+    "All Saint's Day (next Monday)",
+    month=11,
+    day=1,
+    offset=next_monday_offset,
+)
+
+CartagenaIndependenceDay = Holiday(
+    "Cartagena Independence Day",
+    month=11,
+    day=11,
+    observance=next_monday_offset,
+)
+
+ImmaculateConception = immaculate_conception()
+
+ChristmasDay = christmas()
+
+LastTradingDay = new_years_eve(observance=previous_friday)
```

Comment:
Where did you find that new year's eve is observed on the previous Friday? Looking at the official website [here](https://en.bvc.com.co/pps/tibco/portalbvc/Home/Mercados/dianobursatil?action=dummy) and [here](https://www.bvc.com.co/pps/tibco/portalbvc/Home/Mercados/boletines?com.tibco.ps.pagesvc.action=updateRenderState&rp.currentDocumentID=-62968345_16797c61d92_2197c0a84ca9&rp.attachmentPropertyName=Attachment&com.tibco.ps.pagesvc.targetPage=1f9a1c33_132040fa022_-78750a0a600b&com.tibco.ps.pagesvc.mode=resource&rp.redirectPage=1f9a1c33_132040fa022_-787e0a0a600b), I see for example that December 31st, 2018 was a non-trading day, but that is a Monday.

Reply:
I think the "previous_friday" pandas function is poorly named, it means if the date is on a weekend bump it back to friday